sqld/src/replication/replica/injector.rs

```diff
@@ -143,18 +22,25 @@ impl<'a> FrameInjector<'a> {
             hook_ctx,
         )?;
 
-        Ok(Self { conn, ctx })
+        Ok(Self { conn })
     }
 
-    /// sets the injector's frames to the provided frames, trigger a dummy write, and collect the
-    /// injection result.
-    fn inject_frames(&mut self, frames: Frames) -> anyhow::Result<FrameNo> {
-        self.ctx.set_frames(frames);
-
-        let _ = self
-            .conn
-            .execute("create table if not exists __dummy__ (dummy)", ());
-
-        self.ctx.take_result()
+    pub fn step(&mut self) -> anyhow::Result<bool> {
+        let res = self.conn.execute("create table __dummy__ (dummy);", ());
```

Comment:
Previous code had a `IF NOT EXISTS`, I guess it doesn't hurt to keep it? Then we reduce the chance of having to handle some random SQLite error instead of just hitting `Ok(_) => panic!("replication hook was not called"),`, which sounds like a right choice if we failed to run a wal hook.

Reply:
The only path that leads to SQLITE_OK not being called is the hook not being called, which is by design (see later comments about why we return an error).